scenarios/evaluate-safety/evaluate-protected-material-and-indirect-attack-jailbreak.ipynb

```diff
@@ -0,0 +1,512 @@
+    "for output in unfiltered_indirect_attack_outputs:\n",
+    "    for turn in output[\"messages\"]:\n",
+    "        content = turn[\"content\"]\n",
+    "        if isinstance(content, dict):  # user response from callback is dict\n",
+    "            print(f\"{turn['role']} : {content['content'][0:truncation_limit]}\")\n",
+    "        elif isinstance(content, tuple):  # assistant response from callback is tuple\n",
+    "            print(f\"{turn['role']} : {content[0:truncation_limit]}\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from pathlib import Path\n",
+    "\n",
+    "print(unfiltered_indirect_attack_outputs)\n",
+    "print(unfiltered_indirect_attack_outputs.to_eval_qa_json_lines())\n",
+    "output = unfiltered_indirect_attack_outputs.to_eval_qa_json_lines()\n",
+    "xpia_file_path = \"unfiltered_indirect_attack_outputs.jsonl\"\n",
+    "\n",
+    "# Write the output to the file\n",
+    "with Path.open(xpia_file_path, \"w\") as file:\n",
+    "    file.write(output)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now that we have our dataset, we can evaluate it to see if the indirect attacks resulted in jailbreaks. The `IndirectAttackEvaluator` class can take in the dataset and detects instances of jailbreak. Let's use the `evaluate()` API to run the evaluation and log it to our Azure AI Studio Project."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "indirect_attack_eval = IndirectAttackEvaluator(project_scope=project_scope, credential=DefaultAzureCredential())\n",
+    "file_path = \"indirect_attack_outputs.jsonl\"\n",
+    "result = evaluate(\n",
+    "    data=xpia_file_path,\n",
+    "    evaluators={\n",
+    "        \"indirect_attack\": indirect_attack_eval,\n",
+    "    },\n",
+    "    # Optionally provide your AI Studio project information to track your evaluation results in your Azure AI Studio project\n",
+    "    azure_ai_project=project_scope,\n",
+    "    # Optionally provide an output path to dump a json of metric summary, row level data and metric and studio URL\n",
+    "    output_path=\"./mynewindirectattackevalresults.json\",\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can see that our \"model\" application gives us a defect rate broken down by different behaviors resulting from the jailbreak, showing us that we can't deploy our application just yet. Moving forward, to protect our application against indirect jailbreak attacks, we can add an Azure AI Content Safety Prompt Shield which is a mitigation layer to help annotate and block requests to your model or application that contain known indirect attacks for jailbreak. Let's apply this filter and re-run the simulator and evaluation step to see if it helps with our defect rate."
```

Comment:
Please add this link to "Azure AI Content Safety Prompt Shield": https://learn.microsoft.com/en-us/azure/ai-services/content-safety/quickstart-jailbreak